static/app/components/core/layout/stack.mdx

```diff
@@ -0,0 +1,243 @@
+
+```tsx
+<Stack as="section" padding="md" background="primary">
+  Basic stack content
+</Stack>
+```
+
+### Stack Properties
+
+Stack provides a focused set of layout properties: `direction` (defaults to 'column'), `gap`, `justify`, and `align`. These properties influence the layout of its children while maintaining simplicity.
+
+Like other layout components, `Stack` inherits all spacing props like `m`, `p`, `mt`, `mb`, `ml`, `mr`, `pt`, `pb`, `pl`, `pr` and implements responsive props so that the layout can be changed per breakpoint.
+
+#### Column Direction (Default)
+
+<Storybook.Demo>
+  <Stack gap="md" justify="center" align="center" padding="md">
+    <Container padding="md" border="primary" radius="md" background="primary">
+      Item 1
+    </Container>
+    <Container padding="md" border="primary" radius="md" background="primary">
+      Item 2
+    </Container>
+    <Container padding="md" border="primary" radius="md" background="primary">
+      Item 3
+    </Container>
+  </Stack>
+</Storybook.Demo>
+```jsx
+<Stack gap="md" justify="center" align="center" padding="md">
+  <div>Item 1</div>
+  <div>Item 2</div>
+  <div>Item 3</div>
+</Stack>
+```
+
+#### Row Direction
+
+<Storybook.Demo>
+  <Stack direction="row" gap="md" justify="between" align="center" padding="md">
+    <Container padding="md" border="primary" radius="md" background="primary">
+      Item 1
+    </Container>
+    <Container padding="md" border="primary" radius="md" background="primary">
+      Item 2
+    </Container>
+    <Container padding="md" border="primary" radius="md" background="primary">
+      Item 3
+    </Container>
+  </Stack>
+</Storybook.Demo>
+```jsx
+<Stack direction="row" gap="md" justify="between" align="center" padding="md">
+  <div>Item 1</div>
+  <div>Item 2</div>
+  <div>Item 3</div>
+</Stack>
+```
+
+#### Spacing
+
+The `Stack` `gap` property follows the same spacing system as other layout components.
+
+<Storybook.Demo>
+  {['xs', 'sm', 'md', 'lg', 'xl', '2xl'].map(size => (
+    <Stack direction="column" gap="sm" key={size}>
+      <strong>{size} gap</strong>
+      <Stack m="md" gap={size} align="center">
+        <Container padding="md" border="primary" radius="md" background="primary">
+          Item 1
+        </Container>
+        <Container padding="md" border="primary" radius="md" background="primary">
+          Item 2
+        </Container>
+        <Container padding="md" border="primary" radius="md" background="primary">
+          Item 3
+        </Container>
+      </Stack>
+    </Stack>
+  ))}
+</Storybook.Demo>
+```jsx
+<Stack m="md" gap="xs">
+  <div>Item 1</div>
+  <div>Item 2</div>
+  <div>Item 3</div>
+</Stack>
+```
+
+#### Responsive Props
+
+All props support responsive values using breakpoint objects. Breakpoints are: `xs`, `sm`, `md`, `lg`, `xl`, `2xl`.
+
+Example of a responsive stack that uses a static gap, but changes direction based on the breakpoint.
+
+<Storybook.Demo>
```

Comment:
would be cool if that demo could be re-sizable so that can see different directions in action. Could just be a slider that modifies the width of the container. This is what Josh Comeau does for his interactive demos, which I think is really cool, [e.g. here](https://www.joshwcomeau.com/css/container-queries-introduction/).

Reply:
Yeah, interactive demos like that are very helpful for folks to understand how the feature works by trying it out. Let me think about this - we'd probably need container queries for this to work if we wanted to control the width here, but it could be easy to add in other cases where we control something like the number of children or the length of a string